### PR TITLE
[MM-21463] Add origin information to errors in SimpleController

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR adds an `ErrOrigin` field in `UserStatus` with information on the location of errors coming from `SimpleController`.
I've also modified errors string coming from the store to help identify them better.

#### Ticket

https://mattermost.atlassian.net/browse/MM-21463

### CODE REVIEW

File: loadtest/control/simplecontroller/status.go [hunk 1]
```diff
@@ -15,11 +22,20 @@ func (c *SimpleController) newInfoStatus(info string) control.UserStatus {
 }
 
 func (c *SimpleController) newErrorStatus(err error) control.UserStatus {
+	var origin string
+	if pc, file, line, ok := runtime.Caller(1); ok {
+		if f := runtime.FuncForPC(pc); f != nil {
+			if wd, err := os.Getwd(); err == nil {
+				origin = fmt.Sprintf("%s %s:%d", strings.TrimPrefix(f.Name(), pkgPath), strings.TrimPrefix(file, wd+"/"), line)
```
Comment (agnivade) [inline loadtest/control/simplecontroller/status.go]: Let's use wd + os.PathSeparator

File: loadtest/store/memstore/store.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ func (s *MemStore) Channels(teamId string) ([]model.Channel, error) {
 
 func (s *MemStore) SetChannels(channels []*model.Channel) error {
 	if channels == nil {
-		return errors.New("channels shouldn't be nil")
+		return errors.New("memstore: channels shouldn't be nil")
```
Comment (agnivade) [inline loadtest/store/memstore/store.go]: unrelated: I just noticed that elsewhere we use "should not", but here we use "shouldn't". Let's change this to "should not" while we are here. Or feel free to do it in a separate PR too.

File: loadtest/control/status.go [hunk 3]
```diff
@@ -30,4 +30,7 @@ type UserStatus struct {
 	Info string
 	// Err contains the error encountered while performing the action.
 	Err error
```
[the remark this answers is not in the document]
Comment (streamer45) [reply loadtest/control/status.go]: The problem I see doing this is that it becomes far less readable to have the whole error string include the origin info. Having that as a separate piece of metadata in `mlog` does help.

[the remark this answers is not in the document]
Comment (agnivade) [reply loadtest/control/status.go]: Ok, we can still have that by having Error() return the normal stuff and have an `Origin` method to return what we want. And still have a separate Error type.

**Pull request merged**